package ntut.csie.test.EmptyCatchBlock;

import ntut.csie.aspect.AspectDemo;
import java.sql.SQLException;
import ntut.csie.RobustaUtils.AspectJSwitch;

public aspect AspectDemoAspectException {

	before() throws SQLException: (call(* *(..) throws SQLException) ) &&  withincode(* AspectDemo.callStaticMethod(..)){

	String name = thisJoinPoint.getSignature().getName();
	String operation = AspectJSwitch.getInstance().getOperation(name);
	if (operation.equals("f(SQLException)"))
		throw new SQLException("This exception is thrown from EmptyCatchBlock's unit test by using AspectJ.");

	}
	
	Object around():(call(* *(..)) ||call(*.new(..)) ) && withincode (* AspectDemo.*(..)){
		String method = thisJoinPoint.toString();
		if(method.contains("Exception(Throwable)")){
			return proceed();
		}else{
			return null;
		}
	}
	
}